### ISSUE TITLE
[Pending/Scanning] $0 reports are being created when a card transaction is imported

### ISSUE THREAD

**Issue by kevinksullivan** [Expensify/App#48497]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**  https://expensify.testrail.io/index.php?/cases/results/3073299 (@kevinksullivan found this when testing, so it wasn't necessarily during regression testing, but was following these exact instructions)
**Email or phone of affected tester (no customers):** kevin@expensify.com
**Issue reported by:** @kevinksullivan 
**Slack conversation:**  https://expensify.slack.com/archives/C036QM0SLJK/p1725201856655499

## Action Performed:

1. Create a new login that matches your Expensify email login, and append @cardtest.expensify.com (e.g. kevin@cardtest.expensify.com)
2. Create a new Collect workspace and make that the default for your account
5. Get card assigned to yourself (ask @kevin if you don't have a >$0 limit on this account already)
6. Navigate to NewDot in your test account, and go to Settings > Wallet > Expensify Card > Reveal details (on virtual card)
7. Navigate to this [paypal link](https://github.com/Expensify/Expensify/files/3974273/payButtons.html.zip) and select “buy” on the first button on that screen, and use your virtual card details to make the purchase.

## Expected Result:
Get one report with the new pending transaction

## Actual Result:
New $0 reports are showing up. Even after the original batch showed up, I saw another one get created ~30 seconds later. I am now seeing another report preview.

## Platforms:
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![image](https://github.com/user-attachments/assets/2ce02881-ebfd-4e92-b3b4-78f848f10f56)

![image](https://github.com/user-attachments/assets/4c840a03-2c2f-4bdf-b817-e2b08de63af6)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by OfstadC** [comment]
Looks like this is being discussed [here](https://expensify.slack.com/archives/C036QM0SLJK/p1725283899357629). Will follow along there

**Comment by OfstadC** [comment]
Still following the Slack thread. If there is anything you'd like me to do here @kevinksullivan, please let me know 😃

**Comment by trjExpensify** [comment]
https://expensify.slack.com/archives/C9YU7BX5M/p1725889678991439?thread_ts=1725201896.273159&cid=C9YU7BX5M - this is the one to follow with getting QA to repro.

**Comment by OfstadC** [comment]
Any update here @kevinksullivan ? 😃

**Comment by OfstadC** [comment]
I believe Kevin is OoO this week. I _think_ we're just waiting on confirming if this can be reproduced. So I'll see what I can track down 😃

**Comment by trjExpensify** [comment]
Asked a Q about this one in the [update thread](https://expensify.slack.com/archives/C036QM0SLJK/p1726504559450319?thread_ts=1726504556.682499&cid=C036QM0SLJK) today. Seems like perhaps a similar RCA to https://github.com/Expensify/App/issues/49037, but not sure.

**Comment by OfstadC** [comment]
Going to move to Weekly since this is on hold

**Comment by OfstadC** [comment]
Not overdue - onhold for https://github.com/Expensify/App/issues/49037

**Comment by OfstadC** [comment]
Bump @grgia 😄

**Comment by grgia** [comment]
Removing hold, moving to daily

**Comment by grgia** [comment]
Alright let me get testing this on cardtest

**Comment by grgia** [comment]
Currently blocked from logging in on staging due to a deploy blocker, I'm helping look into that since I want to test this flow on staging

**Comment by grgia** [comment]
[Reported here ](https://expensify.slack.com/archives/C06ML6X0W9L/p1729563607850989)

**Comment by grgia** [comment]
# **Test 1:**
## Creating a card transaction

https://github.com/user-attachments/assets/c288db8b-0496-4213-ba8b-6aa1d480ce4b

No $0 report here

**Comment by grgia** [comment]
If I do it on dev via clitools, I can see the flashing 0$ report, but not an extra report

**Comment by grgia** [comment]
I cant reliably reproduce with card transaction creation. Will try company cards next

**Comment by trjExpensify** [comment]
Whatcha' think the problem is?

**Comment by grgia** [comment]
@trjExpensify I can't reproduce this, when bugs come up I think the problem is due to extra reports being created in the Create Transaction flow and then the report is subsequently added to a different report 

https://github.com/user-attachments/assets/5bc4fbda-8b9e-4dfe-abf0-aeec3b28d865

**Comment by grgia** [comment]
Are you able to reproduce this?

**Comment by grgia** [comment]
My theory was that it has to do with this section of code 

https://github.com/Expensify/Web-Expensify/blob/be06410f5e73b80194a450179a5bf88268b5cf09/lib/ReportAPI.php#L3206-L3218

**Comment by grgia** [comment]
Maybe this line wasn't catching all cases where we DONT want to create a new report
https://github.com/Expensify/Web-Expensify/blob/be06410f5e73b80194a450179a5bf88268b5cf09/lib/ReportAPI.php#L3212

**Comment by grgia** [comment]
But that's only a theory, I can't test it if I can't get reproduction steps down
